### PR TITLE
feat: should wait for background task finish on afterEach

### PR DESCRIPTION
**Pull request by fengmk2**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

**Comment by fengmk2** [comment]
依赖 https://github.com/eggjs/egg/pull/2872

**Comment by fengmk2** [comment]
https://github.com/eggjs/egg/pull/2874

File: bootstrap.js [hunk 1]
```diff
@@ -8,7 +8,10 @@ if (process.env.EGG_BASE_DIR) options.baseDir = process.env.EGG_BASE_DIR;
 const app = mock.app(options);
 
 before(() => app.ready());
-afterEach(mock.restore);
+afterEach(() => {
```
Comment (atian25) [inline bootstrap.js]: 话说，这里怎么没有 `after(() => app.close());`

File: app/extend/application.js [hunk 2]
```diff
@@ -422,6 +424,29 @@ module.exports = {
       assert(content.includes(str), `Can't find String:"${str}" in ${filepath}`);
     }
   },
+
+  // private method
+  backgroundTasksFinished() {
```
Comment (atian25) [inline app/extend/application.js]: `onBackgroundTasksFinished` or `waitBackgroundTasksFinished`?

Comment (fengmk2) [reply app/extend/application.js]: `await backgroundTasksFinished()`，还是这个比较好。而且它一般都不会被 app 开发者使用到。

File: bootstrap.js (hunk 1 above)
Comment (dead-horse) [reply bootstrap.js]: 应用层不需要 close

**Comment by fengmk2** [comment]
还差 https://github.com/eggjs/egg/pull/2872

File: lib/app.js [hunk 3]
```diff
@@ -66,6 +67,9 @@ class MockApplication extends EventEmitter {
 
     const Application = bindMessenger(egg.Application, agent);
     const app = this._app = new Application(Object.assign({}, this.options));
+    // egg-mock plugin need to override egg context
+    Object.assign(app.context, context);
```
Comment (popomore) [inline lib/app.js]: egg-mock 本身也是插件

**Comment by fengmk2** [comment]
3.19.0

File: lib/context.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+module.exports = {
+  runInBackground(scope) {
+    const promise = this._runInBackground(scope);
```
Comment (atian25) [inline lib/context.js]: 这地方把 egg@1 搞挂了

**Pull request merged**